compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java

```diff
@@ -14482,55 +14484,88 @@ private STNode parseListMatchPatternMemberRhs() {
     private STNode parseMappingMatchPattern() {
         startContext(ParserRuleContext.MAPPING_MATCH_PATTERN);
         STNode openBraceToken = parseOpenBrace();
-        List<STNode> fieldMatchPatternList = new ArrayList<>();
-        STNode fieldMatchPatternRhs = null;
-        boolean isEndOfFields = false;
+        STNode fieldMatchPatterns = parseFieldMatchPatternList();
+        STNode closeBraceToken = parseCloseBrace();
+        endContext();
+        return STNodeFactory.createMappingMatchPatternNode(openBraceToken, fieldMatchPatterns, closeBraceToken);
+    }
+
+    private STNode parseFieldMatchPatternList() {
+        List<STNode> fieldMatchPatterns = new ArrayList<>();
+        if (isEndOfMappingMatchPattern()) {
+            return STNodeFactory.createEmptyNodeList();
+        }
```

Comment:
We don't need this right? `parseFieldMatchPatternMember()` will return `null` if  peek is `CLOSE_BRACE_TOKEN`. We can also add `EOF_TOKEN` to `parseFieldMatchPatternMember` as well